api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/EptsCalculations.java

```diff
@@ -152,25 +154,42 @@ public static CalculationResultMap lastObs(
   }
 
   /**
-   * Evaluates the first program enrollment of the specified program
+   * Evaluates the last patient state for the specified programWorkflowState
    *
-   * @param program the program
-   * @param cohort the patient ids
-   * @param context the calculation context
-   * @return the enrollments in a calculation result map
+   * @param cohort
+   * @param location
+   * @param startDate
+   * @param endDate
+   * @param programWorkflowState
+   * @param context
+   * @return
    */
-  public static CalculationResultMap firstProgramEnrollment(
+  public static CalculationResultMap allPatientStates(
+      Collection<Integer> cohort,
+      Location location,
+      ProgramWorkflowState programWorkflowState,
+      PatientCalculationContext context) {
+
+    JembiPatientStateDefinition def = new JembiPatientStateDefinition();
+    def.setLocation(location);
+    def.setStartedOnOrBefore(context.getNow());
```

Comment:
patientState startdate being now!!! i think i need to re-look through the requirements.